### PR TITLE
Remove all support for the legacy `buffer_t` type

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Comment by abadams** [comment]
Lots of incorrect usage of the old type in the repo revealed here. If we decide we have to hang onto the legacy support for a bit longer we should at least merge the fixes.

**Comment by steven-johnson** [comment]
Urk. Want me to split this into two PRs (one to fix all existing bogus usage and comments, and another to remove the support entirely)?

**Comment by abadams** [comment]
In theory it would be cleaner, but doesn't seem worth the effort. Hopefully we can just remove support entirely.

**Comment by steven-johnson** [comment]
(Note that the hexagon-related 'buffer_t' type is handled separately, in https://github.com/halide/Halide/pull/4524)

**Comment by steven-johnson** [comment]
Ready for review, but not yet ready to land (I'd prefer to do a little preflight work inside google first).

File: python_bindings/correctness/the_sort_function.c [hunk 1]
```diff
@@ -3,72 +3,24 @@
  * gcc -std=c99 the_sort_function.c -shared -o the_sort_function.so
  */
 
+#include "HalideRuntime"
```
Comment (abadams) [inline python_bindings/correctness/the_sort_function.c]: Does this work without the .h?

**Comment by abadams** [comment]
Deleted code is best code.

File: python_bindings/correctness/the_sort_function.c (hunk 1 above)
Comment (steven-johnson) [reply python_bindings/correctness/the_sort_function.c]: No, but apparently it's not currently actively built and tested :-/

**Pull request merged**